src/compas_invocations2/grasshopper.py

```diff
@@ -0,0 +1,191 @@
+    }
+)
+def yakerize(
+    ctx,
+    manifest_path: str,
+    logo_path: str,
+    gh_components_dir: str = None,
+    readme_path: str = None,
+    license_path: str = None,
+    version: str = None,
+    target_rhino: str = "rh8",
+) -> bool:
+    """Create a Grasshopper YAK package from the current project."""
+    if target_rhino not in ["rh6", "rh7", "rh8"]:
+        invoke.Exit("Invalid target Rhino version. Must be one of: rh6, rh7, rh8")
+
+    gh_components_dir = gh_components_dir or _get_user_object_path(ctx)
+    if not gh_components_dir:
+        invoke.Exit("Please provide the path to the directory containing the .ghuser files.")
+
+    readme_path = readme_path or os.path.join(ctx.base_folder, "README.md")
+    if not os.path.exists(readme_path):
+        invoke.Exit(f"Readme file not found at {readme_path}. Please provide a valid path.")
+
+    license_path = license_path or os.path.join(ctx.base_folder, "LICENSE")
+    if not os.path.exists(license_path):
+        invoke.Exit(f"License file not found at {license_path}. Please provide a valid path.")
+
+    version = version or _get_version_from_toml(os.path.join(ctx.base_folder, "pyproject.toml"))
+    target_dir = os.path.join(ctx.base_folder, "dist", "yak_package")
+
+    #####################################################################
+    # Copy manifest, logo, misc folder (readme, license, etc)
+    #####################################################################
+    # if target dit exists, make sure it's empty
+    if os.path.exists(target_dir) and os.path.isdir(target_dir):
+        _clear_directory(target_dir)
+    else:
+        os.makedirs(target_dir, exist_ok=False)
+
+    manifest_target = shutil.copy(manifest_path, target_dir)
+    _set_version_in_manifest(manifest_target, version)
+    shutil.copy(logo_path, target_dir)
+
+    path_miscdir: str = os.path.join(target_dir, "misc")
+    os.makedirs(path_miscdir, exist_ok=False)
+    shutil.copy(readme_path, path_miscdir)
+    shutil.copy(license_path, path_miscdir)
+
+    for f in os.listdir(gh_components_dir):
+        if f.endswith(".ghuser"):
+            shutil.copy(os.path.join(gh_components_dir, f), target_dir)
+
+    #####################################################################
+    # Yak exe
+    #####################################################################
+
+    # yak executable shouldn't be in the target directory, otherwise it will be included in the package
+    target_parent = os.sep.join(target_dir.split(os.sep)[:-1])
+    try:
+        yak_exe_path = _download_yak_executable(target_parent)
+    except ValueError:
+        invoke.Exit("Failed to download the yak executable")
+    else:
+        yak_exe_path = os.path.abspath(yak_exe_path)
+
+    with chdir(target_dir):
+        try:
+            # not using `ctx.run()` here to get properly formatted output (unicode+colors)
+            os.system(f"{yak_exe_path} build --platform any")
```

Comment:
does `chdir` have any effect if we don't use `ctx.run()`?

Reply:
mmh it appears so yea. yak packs the current directory. but wouldn't it though? `chdir` also just uses `os`